### PR TITLE
load twemoji dynamically as colr or sbix; fix monospace

### PR DESCRIPTION
**Pull request by ara4n**
This uses a stripped down fork of https://github.com/RoelN/ChromaCheck/blob/master/chromacheck.js to determine whether your browser can render COLR fonts or not.  If it can, it dynamically loads twemoji as COLR, failing that sbix.

The sbix version of the font was created by converting the colr one using TransType.

This also makes monospace fonts consistent as Inconsolata, to avoid inconsistencies from depending on system fonts.

If Twemoji fails entierly, we explicitly fall back to Apple or Noto emoji if we have them installed to avoid half-assed emoji in Arial, Helvetica, Courier etc taking priority.

should fix https://github.com/vector-im/riot-web/issues/9651 and https://github.com/vector-im/riot-web/issues/9765

### CODE REVIEW

File: res/themes/light/css/_fonts.scss [hunk 1]
```diff
@@ -33,31 +33,52 @@
...
 
-/*
- * Fira Mono
- * Used for monospace copy, i.e. code
- */
-
+/* latin-ext */
 @font-face {
-    font-family: 'Fira Mono';
-    src: url('$(res)/fonts/Fira_Mono/FiraMono-Regular.ttf') format('truetype');
```
Comment (jryans) [inline res/themes/light/css/_fonts.scss]: Shouldn't we delete the Fira files as well?

File: src/utils/FontManager.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+export async function fixupColorFonts() {
+    if (colrFontSupported !== undefined) {
+        return;
+    }
+
+    // we programatically add the right fontface.
+    let font;
+    if (await isColrFontSupported()) {
+        font = new FontFace("Twemoji", "url('../../fonts/Twemoji_Mozilla/TwemojiMozilla-colr.woff2')", {});
```
Comment (jryans) [inline src/utils/FontManager.js]: You should be able to `require` the font path like we do for images in JS, and that will allow you to keep the cache busting.

**Comment by bwindels** [comment]
addressed the comments so we can get this in the RC asap. Can you have another look @jryans ?

File: res/css/_common.scss [hunk 3]
```diff
@@ -34,6 +34,7 @@ body {
 
 pre, code {
     font-family: $monospace-font-family;
+    font-size: 100% ! important;
```
Comment (jryans) [inline res/css/_common.scss]: Nit: Remove space:

```suggestion
    font-size: 100% !important;
```

We need to update @ara4n's brain as well... 😆

**Comment by bwindels** [comment]
The big emojis appear a bit clipped atm in Chrome on Linux:
image
Don't think this is a blocking issue though.

File: src/utils/FontManager.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+let colrFontSupported = undefined;
+
+async function isColrFontSupported() {
+    if (colrFontSupported !== undefined) {
+        return colrFontSupported;
+    }
+
+    // Firefox has supported COLR fonts since version 26
+    // but doesn't support the check below with content blocking enabled.
```
Comment (jryans) [inline src/utils/FontManager.js]: I am curious what actually goes wrong with the content blocking, but it doesn't seem worth digging into right now...

**Comment by bwindels** [comment]
Tested on FF and Chrome with added changes and without riot-web webpack config PR, working fine.

**Pull request merged**